app/services/handover_date_service.rb

```diff
@@ -8,143 +8,12 @@
-    def policy_case?
-      sentenced_after_policy_started = @offender.sentence_start_date >= policy_start_date
-      release_after_cutoff = release_date && release_date >= policy_cutoff_date
-
-      # Offenders must have been sentenced on/after the OMIC policy start date,
-      # or have a release date which is on/after the 'cutoff' date.
-      within_policy_dates = sentenced_after_policy_started || release_after_cutoff
-
-      if in_open_conditions?
-        # There are additional rules to decide if OMIC open prison rules apply
-        within_policy_dates && open_prison_rules_apply?
-      else
-        within_policy_dates
-      end
-    end
-
-    def earliest_release
-      @earliest_release ||= @offender.earliest_release_for_handover
-    end
-
-    def release_date
-      earliest_release&.date
-    end
-
-    def open_prison_rules_apply?
-      (
-        # Offender falls into the HMP Prescoed open prison pilot
-        @offender.prison_id == PrisonService::PRESCOED_CODE &&
-          @offender.prison_arrival_date >= PRESCOED_POLICY_START_DATE &&
-          welsh_offender?
-      ) ||
-      (
-        # Open prison rules apply because offender arrived after the open prison policy general start date
-        in_male_open_prison? && @offender.prison_arrival_date >= OPEN_PRISON_POLICY_START_DATE
-      ) ||
-      (
-        # Open policy launched before Women's prisons – so no need to check for a 'women's open policy start date'
-        in_womens_prison? && @offender.category_code == 'T'
-      )
-    end
-
-    def in_womens_prison?
-      # TODO: We should be able to use @offender.prison.womens? (much nicer!), but first we'll need to replace use of
-      #       OpenStruct objects in the tests with proper MpcOffender objects
-      PrisonService.womens_prison?(@offender.prison_id)
-    end
-
-    def in_open_conditions?
-      in_male_open_prison? || (in_womens_prison? && @offender.category_code == 'T')
-    end
-
-    def in_male_open_prison?
-      # TODO: We should be able to use @offender.prison.mens_open? (much nicer!), but first we'll need to replace use of
-      #       OpenStruct objects in the tests with proper MpcOffender objects
-      PrisonService.open_prison?(@offender.prison_id)
-    end
-
-    def hub_or_private?
-      PrisonService.english_hub_prison?(@offender.prison_id) ||
-        PrisonService.english_private_prison?(@offender.prison_id)
-    end
-
-    def welsh_offender?
-      @offender.welsh_offender == true
-    end
-
-    def policy_start_date
-      if in_womens_prison?
-        WOMENS_POLICY_START_DATE
-      elsif welsh_offender?
-        WELSH_POLICY_START_DATE
-      else
-        ENGLISH_POLICY_START_DATE
-      end
-    end
-
-    def policy_cutoff_date
-      if in_womens_prison?
-        WOMENS_CUTOFF_DATE
-      elsif welsh_offender?
-        WELSH_CUTOFF_DATE
-      elsif hub_or_private?
-        ENGLISH_PRIVATE_CUTOFF
-      else
-        ENGLISH_PUBLIC_CUTOFF
-      end
-    end
-  end
```

Comment:
_Love_ to see this, I hated that wrapper

Reply:
Yes, to me it was a bit "shove this mess under the carpet" when I want all the mess front and centre in MpcOffender so I can decide how to carve it up best